test/order_optimizer_test.rb

```diff
@@ -75,7 +75,7 @@ def test_that_it_works_with_many_different_pack_sizes
       '1-pack' => { quantity: 1, price_per_unit: 9 },
       '10-pack' => { quantity: 10, price_per_unit: 6 },
       '100-pack' => { quantity: 100, price_per_unit: 4 },
-      '1000-pack' => { quantity: 1000, price_per_unit: 3.9 }
+      '1000-pack' => { quantity: 1000, price_per_unit: BigDecimal("3.9") }
```

Comment:
I think we should convert the price per unit to Decimal inside the Order Optimiser just to be sure